src/server.rs

```diff
@@ -4660,6 +4544,23 @@ impl CompletionVarResult {
     }
 }
 
+impl CompletionVarType {
+    pub fn from_monotype(typ: &MonoType) -> Option<Self> {
```

Comment:
Could this be implemented as a `From` trait? Since the matcher expects to be exhaustive by default, we don't need to worry about the `_ =>` case at all, which means we don't need to return the `Option`, right? Or am I missing something?

Reply:
It needs to return `None` though? So it could be `TryFrom` I suppose.